src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
+     * Pipeline run started event extensions
      */
-    public static CloudEvent createRepositoryEvent(
-            final String repositoryEventType, final String repositoryId,
-            final String repositoryName, final String repositoryURL,
-            final String repositoryData) {
+
+
+    public static CloudEvent createPipelineRunStartedEvent(
+            final String eventType,
+            final String id, final URI source,
+            final String pipelineName, final URI url,
+            final String pipelineRunData) {
         CloudEvent ceToSend =
-                buildCloudEvent(repositoryEventType, repositoryData)
-                .withExtension("repositoryid", repositoryId)
-                .withExtension("repositoryname", repositoryName)
-                .withExtension("repositoryurl", repositoryURL)
-                .build();
+                buildCloudEvent(eventType, pipelineRunData)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("pipelinename", pipelineName)
+                        .withExtension("url", url)
+                        .build();
         return ceToSend;
     }
 
     /**
-     * Creates a continuous delivery build events
+     * Creates a continuous delivery Task run started events
      * using {@link CloudEventBuilder}
      * and returns {@link CloudEvent} object.
      *
-     * @param buildEventType
-     * @param buildId
-     * @param buildName
-     * @param buildArtifactId
-     * @param buildData
+     * @param eventType
+     * @param id
+     * @param source
+     * @param taskName
+     * @param pipelineRun
+     * @param url
+     * @param taskRunData
      * @return the cloudEvent object with continuous delivery
-     * build event extensions
+     * task run started event extensions
      */
-    public static CloudEvent createBuildEvent(
-            final String buildEventType, final String buildId,
-            final String buildName, final String buildArtifactId,
-            final String buildData) {
+
+    public static CloudEvent createTaskRunStartedEvent(
+            final String eventType, final String id,
+            final URI source, final String taskName,
+            final PipelineRun pipelineRun, final URI url,
+            final String taskRunData) {
         CloudEvent ceToSend =
-                buildCloudEvent(buildEventType, buildData)
-                .withExtension("buildid", buildId)
-                .withExtension("buildname", buildName)
-                .withExtension("buildartifactid", buildArtifactId)
-                .build();
+                buildCloudEvent(eventType, taskRunData)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("taskname", taskName)
+                        .withExtension("pipelinerun", pipelineRun.toString())

```

Comment:
please use `jackson-databind` to write this `pipelineRun` as String, say objectMapper.writeValueAsString(pipelineRun)